CRM/Upgrade/Incremental/php/FiveSixtyTwo.php

```diff
@@ -39,6 +39,11 @@ public function setPreUpgradeMessage(&$preUpgradeMessage, $rev, $currentVer = NU
         }
       }
     }
+
+    $activityCount = CRM_Core_DAO::singleValueQuery('SELECT MAX(id) FROM civicrm_activity');
+    if ($activityCount > 0) {
+      $preUpgradeMessage .= '<p>' . ts('Your database contains CiviCase activity revisions which are deprecated and will begin to appear as duplicates in SearchKit/api4/etc.<ul><li>For further instructions see this <a %1>Lab Snippet</a>.</li></ul>', [1 => $docAnchor]) . ts('Note: You might have followed these steps already but unfortunately a previous upgrade which removed CiviCase Activity revisions failed to account for all the places this functionality existed. This means that new Revisions have been added since you removed them.') . '</p>';
```

Comment:
@stesi561 $docAnchor isn't defined here you may need to bring that in

Reply:
I don't want to get bogged down in wording but the upgrade in 5.54 didn't remove revisions, it "attempted to no longer created new revisions". So maybe "...a previous upgrade which started ignoring the setting to create CiviCase Activity revisions failed to account..."